presto-main/src/main/java/io/prestosql/metadata/MetadataListing.java

```diff
@@ -69,10 +70,23 @@ private MetadataListing() {}
 
     public static Set<SchemaTableName> listViews(Session session, Metadata metadata, AccessControl accessControl, QualifiedTablePrefix prefix)
     {
-        Set<SchemaTableName> tableNames = metadata.listViews(session, prefix).stream()
-                .map(QualifiedObjectName::asSchemaTableName)
-                .collect(toImmutableSet());
-        return accessControl.filterTables(session.toSecurityContext(), prefix.getCatalogName(), tableNames);
+        return getViews(session, metadata, accessControl, prefix).keySet();
+    }
+
+    public static Map<SchemaTableName, ConnectorViewDefinition> getViews(Session session, Metadata metadata, AccessControl accessControl, QualifiedTablePrefix prefix)
+    {
+        Map<SchemaTableName, ConnectorViewDefinition> views =
+                metadata.getViews(session, prefix).entrySet().stream()
+                        .collect(toImmutableMap(entry -> entry.getKey().asSchemaTableName(), Entry::getValue));
+
+        Set<SchemaTableName> accessible = accessControl.filterTables(
+                session.toSecurityContext(),
+                prefix.getCatalogName(),
+                views.keySet());
```

Comment:
I don't think you need to break parameters in separate lines, the line is should not be very long